activestorage/lib/active_storage/downloader.rb

```diff
@@ -35,7 +35,7 @@ def download(key, file)
       end
 
       def verify_integrity_of(file, checksum:)
-        unless ActiveStorage.checksum_implementation.file(file).base64digest == checksum
+        unless service.compute_checksum(file) == checksum
           raise ActiveStorage::IntegrityError
```

Comment:
Needs to be a little more clear when raising exception:

```suggestion
        actual_checksum = service.compute_checksum(file) 
        unless actual_checksum == checksum
          raise ActiveStorage::IntegrityError, "Checksum verification failed expecting #{checksum}, but downloaded file having #{actual_checksum}"
```